dex/farm/src/custom_rewards.rs

```diff
@@ -8,25 +8,33 @@ pub trait CustomRewardsModule:
     + farm_token::FarmTokenModule
     + rewards::RewardsModule
 {
-    fn mint_per_block_rewards(&self, token_id: &TokenIdentifier) -> BigUint {
+    #[only_owner]
+    #[payable("*")]
+    #[endpoint(depositRewards)]
+    fn deposit_rewards(&self, #[payment_token] payment_token: TokenIdentifier) -> SCResult<()> {
+        let reward_token_id = self.reward_token_id().get();
+        require!(payment_token == reward_token_id, "Invalid token");
+        Ok(())
+    }
+
+    fn calculate_and_increase_per_block_rewards(&self) -> BigUint {
         let current_block_nonce = self.blockchain().get_block_nonce();
         let last_reward_nonce = self.last_reward_block_nonce().get();
 
         if current_block_nonce > last_reward_nonce {
             let to_mint = self.calculate_per_block_rewards(current_block_nonce, last_reward_nonce);
 
-            if to_mint != 0 {
-                self.send().esdt_local_mint(token_id, 0, &to_mint);
-            }
+            // rewards are not minted, but rather deposited by the owner
```

Comment:
I would put a check for the number of actual remained rewards from sc balance to be grater than the minted rewards for the current_block - last_block period. When we will get to 0, then the SC should itself call end_produce_rewards. Otherwise, any claim rewards transactions would fail with no rewards to send